base/refpointer.jl

```diff
@@ -13,11 +13,20 @@ Creation of a `Ref` to a value `x` of type `T` is usually written `Ref(x)`.
 Additionally, for creating interior pointers to containers (such as Array or Ptr),
 it can be written `Ref(a, i)` for creating a reference to the `i`-th element of `a`.
 
-When passed as a `ccall` argument (either as a `Ptr` or `Ref` type), a `Ref` object will be
-converted to a native pointer to the data it references.
+`Ref{T}()` creates a reference to a value of type `T` without initialization.
+For a bitstype `T`, the value will be whatever currently resides in the memory
+allocated. For a non-bitstype `T`, the value will be `undef` and will result in
```

Comment:
This seems confusing, `undef` represents a well-defined singleton value in Julia. I would just write "undefined". Also, it's not the value which will result in an error, but the act of dereferencing.

Reply:
I corrected this in a49c8cc

Reply:
Sorry for the delay. If I'm not mistaken, my second point was not addressed. That is, the reference will not "result in an error"; trying to dereference it will result in an error.

Reply:
I revised the text in 2b9a7a7 to clarify that attempted dereferencing will cause an error. Thanks @rfourquet